Source/Particles/Collision/BinaryCollision/LinearBreitWheeler/SingleLinearBreitWheelerCollisionEvent.H

```diff
@@ -0,0 +1,176 @@
+ * If the electron-positron pair is produced, the mask is set to true for that given pair
+ * of macrophotons and the weight of the produced particles is stored in p_pair_reaction_weight.
+ *
+ * Note that the word <<pair>> sometimes refers to the electron-positron products of the collision,
+ * sometimes to the macrophotons that are paired to collide in a cell according to the algorithm.
+ * It should be clear from the context.
+ *
+ * @tparam index_type type of the index argument
+ * @param[in] u1x,u1y,u1z momentum components of the first colliding particle
+ * @param[in] u2x,u2y,u2z momentum components of the second colliding particle
+ * @param[in] w1,w2 effective weight of the colliding particles
+ * @param[in] dt is the time step length between two collision calls.
+ * @param[in] dV is the volume of the corresponding cell.
+ * @param[in] pair_index is the index of the colliding pair
+ * @param[out] p_mask is a mask that will be set to true if pair (electron-positron) production
+ * occurs for that pair of macrophotons
+ * @param[out] p_pair_reaction_weight stores the weight of the product particles
+ * @param[in] event_multiplier factor used to increase the number of pair-production events by
+ * decreasing the weight of the produced particles
+ * @param[in] multiplier_ratio factor used to take into account unsampled pairs (i.e. the fact
+ * that a particle only collides with one or few particles of the other species)
+ * @param[in] probability_threshold probability threshold above which we decrease the event
+ * multiplier
+ * @param[in] probability_target_value if the probability threshold is exceeded, this is used
+ * to determine by how much the event multiplier is reduced
+ * @param[in] engine the random engine.
+ */
+template <typename index_type>
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void SingleLinearBreitWheelerCollisionEvent (const amrex::ParticleReal& u1x, const amrex::ParticleReal& u1y,
+                                       const amrex::ParticleReal& u1z, const amrex::ParticleReal& u2x,
+                                       const amrex::ParticleReal& u2y, const amrex::ParticleReal& u2z,
+                                       amrex::ParticleReal w1, amrex::ParticleReal w2,
+                                       const amrex::Real& dt, const amrex::ParticleReal& dV, const int& pair_index,
+                                       index_type* AMREX_RESTRICT p_mask,
+                                       amrex::ParticleReal* AMREX_RESTRICT p_pair_reaction_weight,
+                                       const amrex::ParticleReal& event_multiplier,
+                                       const int& multiplier_ratio,
+                                       const amrex::ParticleReal& probability_threshold,
+                                       const amrex::ParticleReal& probability_target_value,
+                                       const amrex::RandomEngine& engine)
+{
```

Comment:
Overall, it seems that this function could be made simpler by reusing the same helper function `BinaryCollisionUtils::get_collision_parameters` as here:
https://github.com/BLAST-WarpX/warpx/blob/development/Source/Particles/Collision/BinaryCollision/NuclearFusion/SingleNuclearFusionEvent.H#L71
(This function was introduced in #4125 ; see esp: https://github.com/BLAST-WarpX/warpx/pull/4125/files#diff-27f511292ce4272f365209c52010c9767c3c3ddb0ea5af3a6a81393afcdb7b23)

Reply:
It should probably be a follow-up PR:
- Make sure that `get_collision_parameters` is valid for massless particles
- Use it in linear Breit-Wheeler